pkg/node/address.go

```diff
@@ -295,16 +295,28 @@ func AutoComplete(directRoutingDevice string) error {
 // ValidatePostInit validates the entire addressing setup and completes it as
 // required
 func ValidatePostInit() error {
-	if option.Config.EnableIPv4 || option.Config.TunnelingEnabled() {
-		if GetIPv4() == nil {
-			return fmt.Errorf("external IPv4 node address could not be derived, please configure via --ipv4-node")
+	if option.Config.EnableIPv4 && GetIPv4() == nil {
+		return fmt.Errorf("external IPv4 node address could not be derived, please configure via --ipv4-node")
+	}
+
+	if option.Config.EnableIPv6 && GetIPv6() == nil {
+		return fmt.Errorf("external IPv6 node address could not be derived, please configure via --ipv6-node")
+	}
+
+	if option.Config.TunnelingEnabled() {
+		if GetIPv4() == nil && GetIPv6() == nil {
+			return fmt.Errorf("external IPv4 and IPv6 node addresses could not be derived, please configure via --ipv4-node or --ipv6-node")
```

Comment:
Can you elaborate a bit more? Cilium currently doesn't support tunneling over IPv6 ([background](https://github.com/cilium/cilium/issues/17240)), hence why we are enforcing an IPv4 address here.